### DISCUSSION TITLE
Errors seem to clear themselves out when using async values

### DISCUSSION THREAD

**Discussion by ernstoarllano** [react-hook-form/react-hook-form#10809] [Q&A]
For some reason when I try using async values errors don't seem to stick, they seem to clear out immediately:

```
const methods = useForm<InviteLinkFormSchema>({
    resolver: zodResolver(schema),
    values: {
      email_domains: '',
      add_to_groups: '',
      expires_at: defaultExpiresAt(),
      language: defaultLanguage,
      name: defaultName(),
    },
});
```
My form is a multi-step form and triggers validation for each step so for video example it goes like so:

```
const handleNext = () => {
    sanitizeDomains();

    trigger('email_domains').then((isValid) => {
      if (isValid) nextStep();
    });
 };
```
If I switch the form to use `defaultValues` the errors seem to work. Any idea what would cause the errors to clear out? I'm using `v7.45.4` of `react-hook-form` and `v3.2.0` of `@hookform/resolvers`.

https://github.com/react-hook-form/react-hook-form/assets/5606227/0da5bd3d-3d39-4777-a4e8-559ff7815b63

**Comment by leapful** [comment]
@ernstoarllano 

Please provide codesandbox url to re-produce the issue.
